src/iter.rs

```diff
@@ -0,0 +1,685 @@
+use super::{
+    Decoder, DecoderError, DecoderErrorInt, MalformedPacket, TimestampDataRelation, TracePacket,
+};
+
+pub use cortex_m::peripheral::itm::LocalTimestampOptions;
+use std::io::Read;
+
+pub struct Singles<'a, R>
+where
+    R: Read,
+{
+    decoder: &'a mut Decoder<R>,
+}
+
+impl<'a, R> Singles<'a, R>
+where
+    R: Read,
+{
+    pub(super) fn new(decoder: &'a mut Decoder<R>) -> Self {
+        Self { decoder }
+    }
+}
+
+impl<'a, R> Iterator for Singles<'a, R>
+where
+    R: Read,
+{
+    type Item = Result<TracePacket, DecoderError>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let trace = self.decoder.next_single();
+
+        match trace {
+            Err(DecoderErrorInt::Eof) => None,
+            Err(DecoderErrorInt::Io(io)) => Some(Err(DecoderError::Io(io))),
+            Err(DecoderErrorInt::MalformedPacket(m)) => Some(Err(DecoderError::MalformedPacket(m))),
+            Ok(trace) => Some(Ok(trace)),
+        }
+    }
+}
+
+pub type DateTime = chrono::DateTime<chrono::Utc>;
+
+#[derive(Clone)]
+pub struct TimestampConfiguration {
+    /// Frequency of the ITM timestamp clock. Necessary to calculate
+    /// the offset continuously added to [Self::baseline].
+    pub clock_frequency: u32,
+
+    /// Prescaler used for the ITM timestamp clock. Necessary to
+    /// calculate the offset continuously added to [Self::baseline].
+    pub lts_prescaler: LocalTimestampOptions,
```

Comment:
The enum contains information on whether local timestamps should be enabled, and what prescaler to use. For example, `EnabledDiv16`, but also `Disabled`.